### DISCUSSION TITLE
Is it possible to store translations in the database instead of in PO files?

### DISCUSSION THREAD

**Discussion by craftyweb** [OrchardCMS/OrchardCore#16653] [Q&A]
Hi,

Because we need a fast way to add translations to our websites we created a module to add the translations in the dashboard.
We've created a TranslationProvider that handles this and the translations are correctly added to the CultureDictionary. 

However, when we try to translate this in our liquid template the original is showed instead of the translation. We would like to know if something like this is possible and if we need to do some additional steps in order to make this work.

Thanks!

**Comment by craftyweb** [comment]
The problem was with the key that we used to add it to the dictionary. When I've added it with null as context it works

**Comment by hishamco** [comment]
Why not?!! you can create your translation provider, I already did something similar in Data Localization module in OrchardCoreContrib (OCC) 

https://github.com/OrchardCoreContrib/OrchardCoreContrib/tree/main/src/OrchardCoreContrib.Localization/Data
https://github.com/OrchardCoreContrib/OrchardCoreContrib.Modules/tree/main/src/OrchardCoreContrib.DataLocalization
